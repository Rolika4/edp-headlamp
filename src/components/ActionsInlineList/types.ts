import { KubeObjectAction } from '../../types/actions';

export interface ActionsInlineListProps {
  actions: KubeObjectAction[];
}
