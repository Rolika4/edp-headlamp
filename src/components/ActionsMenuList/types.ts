import { KubeObjectAction } from '../../types/actions';

export interface ActionsMenuListProps {
  actions: KubeObjectAction[];
  handleCloseActionsMenu?(): void;
  anchorEl: HTMLElement | null;
}
