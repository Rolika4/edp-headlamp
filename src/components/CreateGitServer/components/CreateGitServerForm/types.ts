import { EDPGitServerKubeObjectInterface } from '../../../../k8s/EDPGitServer/types';
import { DeepPartial } from '../../../../types/global';
import { EDPKubeObjectInterface } from '../../../../types/k8s';

export interface CreateGitServerFormProps {
    editorOpen: boolean;
    setEditorOpen(boolean): void;
    setDialogOpen(boolean): void;
    handleApply(
        gitServerData: DeepPartial<EDPGitServerKubeObjectInterface>,
        gitServerSecretData: DeepPartial<EDPKubeObjectInterface>
    ): void;
    isApplying: boolean;
}
