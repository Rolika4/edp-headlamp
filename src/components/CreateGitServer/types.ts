export interface CreateGitServerProps {
    createDialogOpen?: boolean;
    onClose?(): void;
    setCreateDialogOpen?(boolean): void;
}
