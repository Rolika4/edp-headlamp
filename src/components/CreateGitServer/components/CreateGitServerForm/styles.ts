import { MuiCore } from '../../../../plugin.globals';
import { rem } from '../../../../utils/styling/rem';

const { makeStyles } = MuiCore;

export const useStyles = makeStyles((theme: DefaultTheme) => ({
    form: {
        width: '100%',
        position: 'relative',
        paddingRight: rem(16),

        '& form': {
            display: 'flex',
            flexDirection: 'column',
            justifyContent: 'space-between',
            height: '100%',
        },
    },
    formInner: {
        flex: `1 0 calc(100% - ${rem(75)})`,
        minHeight: rem(100),
        padding: `${rem(16)} ${rem(24)}`,
        overflowY: 'auto',
    },
    actions: {
        flexGrow: 1,
        display: 'flex',
        alignItems: 'center',
        gap: rem(20),
        padding: rem(20),
        backgroundColor: theme.palette.background.paper,
        boxShadow: `0 ${rem(-5)} ${rem(54)} ${rem(-28)} rgba(0,0,0, 0.38)`,
    },
}));
