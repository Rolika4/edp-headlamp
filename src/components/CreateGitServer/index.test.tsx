/**
 * @jest-environment jsdom
 */

import { describe, jest } from '@jest/globals';
import { createTheme } from '@material-ui/core';
import { ThemeProvider } from '@material-ui/styles';
import { configureStore } from '@reduxjs/toolkit';
import { render, screen, waitFor } from '@testing-library/react';
import { SnackbarProvider } from 'notistack';
import React from 'react';
import { Provider } from 'react-redux';
import { namespacesMock } from '../../hooks/useNamespaces/mocks/namespaces.mock';
import { pluginLib } from '../../plugin.globals';
import { CreateGitServer } from './index';
import { CreateGitServerProps } from './types';

const { ApiProxy } = pluginLib;

describe('CreateGitServer', () => {
    it('should render correctly', async () => {
        const props: CreateGitServerProps = {
            createDialogOpen: true,
            setCreateDialogOpen: () => {},
            onClose: () => {},
        };
        const store = configureStore({
            reducer: () => ({}),
        });

        const theme: DefaultTheme = createTheme();

        jest.spyOn(ApiProxy, 'request').mockImplementation(url => {
            if (url.includes('/api/v1/namespaces')) {
                return Promise.resolve(namespacesMock);
            }
        });

        render(
            <Provider store={store}>
                <SnackbarProvider>
                    <ThemeProvider theme={theme}>
                        <CreateGitServer {...props} />
                    </ThemeProvider>
                </SnackbarProvider>
            </Provider>
        );

        await waitFor(() => {
            expect(screen.getByTestId('create-git-server')).toMatchSnapshot();
        });
    });
});
