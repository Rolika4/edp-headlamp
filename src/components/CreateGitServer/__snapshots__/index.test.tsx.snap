// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateGitServer should render correctly 1`] = `
<div
  class="makeStyles-dialog-2"
  data-testid="create-git-server"
>
  <div
    class="makeStyles-dialogTitle-3"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5"
    >
      Create Git Server
    </h5>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeSmall"
        >
          <span />
        </span>
        Edit YAML
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiDialogContent-root makeStyles-dialogContent-4"
  >
    <div
      class="makeStyles-form-5"
    >
      <form>
        <div
          class="makeStyles-formInner-6"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              style="display: flex;"
            >
              <svg
                display="none"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <symbol
                  id="gerrit"
                  viewBox="140.65 274.54 565.46 582.26"
                >
                  <g
                    fill="#fff"
                  >
                    <path
                      d="M439.785 313.992s13.404-8.84 27.002-17.172c13.598-8.332 22.232-21.276 22.232-21.276l34.28 28.147s-16.064 15.553-24.774 21.689c-3.709 2.613-10.946 9.158-10.946 9.158s-14.362-10.077-22.362-13.522c-8.077-3.479-25.432-7.024-25.432-7.024z"
                      stroke="#000"
                      stroke-width="2"
                    />
                    <g
                      stroke="red"
                    >
                      <path
                        d="M464.455 303.746s9.964-5.281 14.324-8.8c4.173-3.37 11.127-11.624 11.127-11.624l25.535 21.325s-6.59 7.627-10.443 10.84c-4.557 3.798-14.83 9.839-14.83 9.839z"
                        stroke-width="3"
                      />
                      <path
                        d="M483.278 312.028s5.495-3.914 7.974-6.19c2.554-2.345 7.057-7.642 7.057-7.642"
                        stroke-width="4"
                      />
                    </g>
                  </g>
                  <path
                    d="M308.44 348.89s4.335-19.284 10.707-26.27c7.428-8.145 19.072-14.154 29.35-15.24 27.818-2.939 63.237 6.19 63.237 6.19s-30.804.727-45.602 4.514c-5.192 1.329-14.794 6.298-14.794 6.298s-16.117.837-29.897 12.093c-13.78 11.255-3.034 9.783-3.034 9.783"
                    fill="#fff"
                    stroke="#000"
                    stroke-width="2"
                  />
                  <path
                    d="m370.64 502.36 3.066 30.484s-16.348-5.082-40.325 4.973c-23.978 10.055-39.219 26.404-39.219 26.404l7.89-.798s20.482-14.437 31.961-17.129c11.48-2.692 23.62-2.839 23.62-2.839s-25.492 10.026-30.75 12.376c-6.258 2.797-15.087 9.2-15.087 9.2l10.476-.297c17.687-7.253 41.322-16.02 59.968-15.047 15.592 1.127 33.94 16.754 33.94 16.754l-3.817-9.99s-10.83-8.193-16.387-13.749c-5.555-5.556-5.555-31.315-5.555-31.315-14.31-2.835-2.117-7.942-19.781-9.027zm83.29 37.5-7.857 35.714s-14.582 1.041-37.482 13.354c-22.9 12.312-39.396 29.513-39.396 29.513l9.196-2.081s16.958-15.308 28.126-19.091 25.28-6.127 25.28-6.127-24.412 12.43-29.419 15.274c-5.96 3.386-13.632 11.228-13.632 11.228l9.898-1.923s6.992-5.932 21.043-11.908c16.942-7.207 34.116-16.118 56.417-14.962 10.43.54 17.37 2.463 17.37 2.463l-4.53-4.99s-6.136-3.645-9.53-4.613c-4.028-1.149-12.495-1.314-12.495-1.314l8.225-36.545c-14.31-2.836-3.55-2.907-21.214-3.992z"
                    fill="#ff0"
                    stroke="#000"
                    stroke-width="2"
                  />
                  <path
                    d="M199.68 396.07c9.838-20.536 27.634-38.953 48.47-48.136 22.732-10.018 51.158-10.088 74.438-3.594 7.694 6.5 10.312-17.18 46.906-27.421 36.594-10.24 91.643-6.51 142.15 22.785 54.049 31.349-23.739 41.92-8.081 94.449 32.384 113.58-63.135 101.52-63.135 101.52s-25.254-16.162-35.355-17.173c-10.102-1.01-46.643-25.125-68.866-28.661-22.223-3.536-36.9 18.622-107.99-9.928-24.318-9.766-33.84-33.335-33.84-33.335s-2.01-35.241 5.303-50.508z"
                    fill="#00f"
                    stroke="#000"
                    stroke-width="2"
                  />
                  <path
                    d="M240.55 484.81s33.988-7.904 57.222-14.975 36.164-43.747 71.519-48.798c35.355-5.05 68.528 22.162 80.65 47.416s50.67 55.619 50.67 55.619 1.01 11.112-14.142 18.183-10.499 7.663-10.499 7.663-43.11-11.414-62.396-21.414-61.236-23.044-92.938-30.796c-26.265-6.423-35.033 2.525-80.086-12.898z"
                    fill="#d45500"
                    stroke="#000"
                    stroke-width="2"
                  />
                  <path
                    d="M530.91 517.48s13.894 99 33.683 157.85c19.789 58.851 78.446 180.47 78.446 180.47l28.103-31.852-52.563-109.72 60.515 95.902 26.012-35.727s-70.663-88.488-93.895-140.46-37.495-152.91-37.495-152.91z"
                    fill="#00f"
                    stroke="#000"
                    stroke-width="2"
                  />
                  <path
                    d="M420 328.79s-40.714 5-57.143 42.143 50.714 115.71 60.714 132.14c10 16.429 24.286 38.571 24.286 38.571s47.677 32.034 82.906 53.844c32.657 20.218 65.227 33.111 65.227 33.111s1.169-.494 7.026-19.196c-12.122-21.718-38.322-53.254-54.504-69.192 24.28 14.117 43.346 28.125 60.104 56.548 0 0 4.96-21.392 6.566-30.789-3.035-9.576-47.477-56.569-47.477-56.569 20.552 8.761 33.751 23.026 50.09 44.773 0 0 3.326-21.56 3.836-32.947-2.525-6.566-33.723-35.059-33.723-35.059 13.165 6.637 19.714 11.066 34.413 24.048 0 0-3.036-60.714-66.607-130.71-63.58-69.98-135.72-50.7-135.72-50.7z"
                    fill="navy"
                    stroke="#000"
                    stroke-width="2"
                  />
                  <path
                    d="M334.089 401.203S305.226 370.59 273.3 375.4c-31.925 4.81-41.108 33.674-39.796 39.796 1.312 6.122 13.557 11.808 22.303 15.744s32.799-.437 45.482-6.56c12.682-6.122 32.799-23.178 32.799-23.178z"
                    fill="red"
                    stroke="#000"
                    stroke-width="2"
                  />
                  <path
                    d="M283.079 379.643c12.502 2.186 20.865 14.094 18.679 26.596-2.187 12.502-14.094 20.865-26.596 18.679s-20.865-14.094-18.68-26.596z"
                    fill="#ff0"
                    stroke="#000"
                    stroke-width="3"
                  />
                  <path
                    d="M284.16 407.54c0 7.12-5.773 12.892-12.892 12.892s-12.892-5.771-12.892-12.891 5.772-12.892 12.892-12.892 12.891 5.772 12.891 12.892z"
                  />
                  <path
                    d="M171.05 416.44c-32.325 20.708-31.452 56.084-27.412 60.63 4.04 4.546 8.081 1.01 15.657-1.515 7.576-2.526 26.264-10.102 42.426-11.112 16.162-1.01 21.213 3.03 23.739-.505 2.525-3.536 4.04-15.152 3.03-25.76-1.01-10.606-1.01-23.233-7.07-25.253-6.061-2.02-38.249-5.072-50.37 3.514z"
                    fill="#ff0"
                    stroke="#000"
                    stroke-width="2"
                  />
                  <path
                    d="M311.92 342.02s12.143 21.071 67.143 32.5 125.71-38.214 125.71-38.214l38.786 38.643s-113.07 25.643-150.21 21c-37.143-4.643-81.429-50.357-81.429-50.357"
                    fill="#fff"
                    stroke="#000"
                    stroke-width="2"
                  />
                  <path
                    d="m473.2 360.596 31.535-16.802 29.107 27.857-37.172 8.71z"
                    fill="#fff"
                    stroke="green"
                    stroke-width="3"
                  />
                  <path
                    color="#000"
                    d="m456.92 379.87-3.576 2.317L471.01 397.2l3.429-2.17z"
                    fill="green"
                    style="text-indent: 0; text-transform: none;"
                    transform="translate(39.02 -24.474)"
                  />
                  <path
                    color="#000"
                    d="m473.33 381.73-21.616 10.277 2.969 2.88 21.438-10.067z"
                    fill="green"
                    style="text-indent: 0; text-transform: none;"
                    transform="translate(39.02 -24.474)"
                  />
                  <path
                    d="M215.62 371.23s-6.19 7.292-8.793 11.289c-2.862 4.394-7.447 13.857-7.447 13.857s49.867-24.083 67.607-30.297c18.279-6.403 54.803-21.787 54.803-21.787s-3.845-1.888-5.91-2.387c-3.102-.75-9.535-.868-9.535-.868s-25.784-.124-42.803 6.134c-17.02 6.258-47.922 24.06-47.922 24.06z"
                    fill="#fff"
                    stroke="#000"
                    stroke-width="2"
                  />
                </symbol>
                <symbol
                  id="gitlab"
                  viewBox="93.97 97.52 192.05 184.99"
                >
                  <path
                    d="m282.83 170.73-.27-.69-26.14-68.22a6.81 6.81 0 0 0-2.69-3.24 7 7 0 0 0-8 .43 7 7 0 0 0-2.32 3.52l-17.65 54h-71.47l-17.65-54a6.86 6.86 0 0 0-2.32-3.53 7 7 0 0 0-8-.43 6.87 6.87 0 0 0-2.69 3.24L97.44 170l-.26.69a48.54 48.54 0 0 0 16.1 56.1l.09.07.24.17 39.82 29.82 19.7 14.91 12 9.06a8.07 8.07 0 0 0 9.76 0l12-9.06 19.7-14.91 40.06-30 .1-.08a48.56 48.56 0 0 0 16.08-56.04Z"
                    style="fill: #e24329;"
                  />
                  <path
                    d="m282.83 170.73-.27-.69a88.3 88.3 0 0 0-35.15 15.8L190 229.25c19.55 14.79 36.57 27.64 36.57 27.64l40.06-30 .1-.08a48.56 48.56 0 0 0 16.1-56.08Z"
                    style="fill: #fc6d26;"
                  />
                  <path
                    d="m153.43 256.89 19.7 14.91 12 9.06a8.07 8.07 0 0 0 9.76 0l12-9.06 19.7-14.91S209.55 244 190 229.25c-19.55 14.75-36.57 27.64-36.57 27.64Z"
                    style="fill: #fca326;"
                  />
                  <path
                    d="M132.58 185.84A88.19 88.19 0 0 0 97.44 170l-.26.69a48.54 48.54 0 0 0 16.1 56.1l.09.07.24.17 39.82 29.82L190 229.21Z"
                    style="fill: #fc6d26;"
                  />
                </symbol>
                <symbol
                  id="github"
                  viewBox="0 0 30 30"
                >
                  <path
                    d="M15 3C8.373 3 3 8.373 3 15c0 5.623 3.872 10.328 9.092 11.63a1.751 1.751 0 0 1-.092-.583v-2.051h-1.508c-.821 0-1.551-.353-1.905-1.009-.393-.729-.461-1.844-1.435-2.526-.289-.227-.069-.486.264-.451.615.174 1.125.596 1.605 1.222.478.627.703.769 1.596.769.433 0 1.081-.025 1.691-.121.328-.833.895-1.6 1.588-1.962-3.996-.411-5.903-2.399-5.903-5.098 0-1.162.495-2.286 1.336-3.233-.276-.94-.623-2.857.106-3.587 1.798 0 2.885 1.166 3.146 1.481A8.993 8.993 0 0 1 15.495 9c1.036 0 2.024.174 2.922.483C18.675 9.17 19.763 8 21.565 8c.732.731.381 2.656.102 3.594.836.945 1.328 2.066 1.328 3.226 0 2.697-1.904 4.684-5.894 5.097C18.199 20.49 19 22.1 19 23.313v2.734c0 .104-.023.179-.035.268C23.641 24.676 27 20.236 27 15c0-6.627-5.373-12-12-12z"
                  />
                </symbol>
              </svg>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  style="display: flex;"
                >
                  <div
                    class="MuiFormControl-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      >
                        <span
                          class="makeStyles-labelWrap-12"
                        >
                          <span
                            class="MuiTypography-root makeStyles-label-13 MuiTypography-body1"
                          >
                            Git provider
                          </span>
                          <span />
                        </span>
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      >
                        <div
                          class="MuiFormGroup-root makeStyles-radioGroup-8 MuiFormGroup-row"
                          role="radiogroup"
                        >
                          <button
                            class="MuiButtonBase-root makeStyles-radioControlButton-9"
                            tabindex="0"
                            type="button"
                          >
                            <label
                              class="MuiFormControlLabel-root makeStyles-radioControlLabel-11"
                            >
                              <span
                                aria-disabled="false"
                                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                              >
                                <span
                                  class="MuiIconButton-label"
                                >
                                  <input
                                    class="PrivateSwitchBase-input-17"
                                    name="gitProvider"
                                    type="radio"
                                    value="gerrit"
                                  />
                                  <span
                                    style="display: block; line-height: 0;"
                                  >
                                    <svg
                                      height="20"
                                      width="20"
                                    >
                                      <use
                                        xlink:href="#gerrit"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </span>
                              <span
                                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                              >
                                Gerrit
                              </span>
                            </label>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </button>
                          <button
                            class="MuiButtonBase-root makeStyles-radioControlButton-9"
                            tabindex="0"
                            type="button"
                          >
                            <label
                              class="MuiFormControlLabel-root makeStyles-radioControlLabel-11"
                            >
                              <span
                                aria-disabled="false"
                                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                              >
                                <span
                                  class="MuiIconButton-label"
                                >
                                  <input
                                    class="PrivateSwitchBase-input-17"
                                    name="gitProvider"
                                    type="radio"
                                    value="gitlab"
                                  />
                                  <span
                                    style="display: block; line-height: 0;"
                                  >
                                    <svg
                                      height="20"
                                      width="20"
                                    >
                                      <use
                                        xlink:href="#gitlab"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </span>
                              <span
                                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                              >
                                Gitlab
                              </span>
                            </label>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </button>
                          <button
                            class="MuiButtonBase-root makeStyles-radioControlButton-9"
                            tabindex="0"
                            type="button"
                          >
                            <label
                              class="MuiFormControlLabel-root makeStyles-radioControlLabel-11"
                            >
                              <span
                                aria-disabled="false"
                                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-14 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                              >
                                <span
                                  class="MuiIconButton-label"
                                >
                                  <input
                                    class="PrivateSwitchBase-input-17"
                                    name="gitProvider"
                                    type="radio"
                                    value="github"
                                  />
                                  <span
                                    style="display: block; line-height: 0;"
                                  >
                                    <svg
                                      height="20"
                                      width="20"
                                    >
                                      <use
                                        xlink:href="#github"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </span>
                              <span
                                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                              >
                                Github
                              </span>
                            </label>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <span
                    class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorError"
                  />
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              style="display: flex;"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
                  style="display: flex;"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      style="display: flex;"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                        >
                          <span
                            class="makeStyles-labelWrap-12"
                          >
                            <span
                              class="MuiTypography-root makeStyles-label-13 MuiTypography-body1"
                            >
                              Host
                            </span>
                          </span>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                          style="display: flex; align-items: flex-end;"
                        >
                          <div
                            class="MuiFormControl-root MuiFormControl-fullWidth"
                          >
                            <div
                              class="MuiFormControl-root MuiTextField-root"
                            >
                              <div
                                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                              >
                                <input
                                  aria-invalid="false"
                                  class="MuiInputBase-input MuiInput-input"
                                  name="gitHost"
                                  placeholder="host-name.com"
                                  type="text"
                                  value=""
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
                  style="display: flex;"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      style="display: flex;"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                        >
                          <span
                            class="makeStyles-labelWrap-12"
                          >
                            <span
                              class="MuiTypography-root makeStyles-label-13 MuiTypography-body1"
                            >
                              User
                            </span>
                            <span />
                          </span>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                          style="display: flex; align-items: flex-end;"
                        >
                          <div
                            class="MuiFormControl-root MuiFormControl-fullWidth"
                          >
                            <div
                              class="MuiFormControl-root MuiTextField-root"
                            >
                              <div
                                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                              >
                                <input
                                  aria-invalid="false"
                                  class="MuiInputBase-input MuiInput-input"
                                  name="gitUser"
                                  type="text"
                                  value="git"
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              style="display: flex;"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
                  style="display: flex;"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      style="display: flex;"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                        >
                          <span
                            class="makeStyles-labelWrap-12"
                          >
                            <span
                              class="MuiTypography-root makeStyles-label-13 MuiTypography-body1"
                            >
                              SSH port
                            </span>
                          </span>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                          style="display: flex; align-items: flex-end;"
                        >
                          <div
                            class="MuiFormControl-root MuiFormControl-fullWidth"
                          >
                            <div
                              class="MuiFormControl-root MuiTextField-root"
                            >
                              <div
                                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                              >
                                <input
                                  aria-invalid="false"
                                  class="MuiInputBase-input MuiInput-input"
                                  name="sshPort"
                                  placeholder="Enter SSH port"
                                  type="number"
                                  value="22"
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
                  style="display: flex;"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      style="display: flex;"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                        >
                          <span
                            class="makeStyles-labelWrap-12"
                          >
                            <span
                              class="MuiTypography-root makeStyles-label-13 MuiTypography-body1"
                            >
                              HTTPS port
                            </span>
                          </span>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                          style="display: flex; align-items: flex-end;"
                        >
                          <div
                            class="MuiFormControl-root MuiFormControl-fullWidth"
                          >
                            <div
                              class="MuiFormControl-root MuiTextField-root"
                            >
                              <div
                                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                              >
                                <input
                                  aria-invalid="false"
                                  class="MuiInputBase-input MuiInput-input"
                                  name="httpsPort"
                                  placeholder="Enter HTTPS port"
                                  type="number"
                                  value="443"
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              style="display: flex;"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  style="display: flex;"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <span
                        class="makeStyles-labelWrap-12"
                      >
                        <span
                          class="MuiTypography-root makeStyles-label-13 MuiTypography-body1"
                        >
                          Access token
                        </span>
                      </span>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      style="display: flex; align-items: flex-end;"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth"
                      >
                        <div
                          class="MuiFormControl-root MuiTextField-root"
                        >
                          <div
                            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
                          >
                            <textarea
                              aria-invalid="false"
                              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
                              name="token"
                              rows="4"
                              style="height: 0px; overflow: hidden;"
                            />
                            <textarea
                              aria-hidden="true"
                              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
                              readonly=""
                              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
                              tabindex="-1"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              style="display: flex;"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  style="display: flex;"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <span
                        class="makeStyles-labelWrap-12"
                      >
                        <span
                          class="MuiTypography-root makeStyles-label-13 MuiTypography-body1"
                        >
                          Private SSH key
                        </span>
                      </span>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      style="display: flex; align-items: flex-end;"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth"
                      >
                        <div
                          class="MuiFormControl-root MuiTextField-root"
                        >
                          <div
                            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline"
                          >
                            <textarea
                              aria-invalid="false"
                              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
                              name="sshPrivateKey"
                              placeholder="-----BEGIN OPENSSH PRIVATE KEY-----
                PRIVATE KEY   
-----END OPENSSH PRIVATE KEY-----"
                              rows="4"
                              style="height: 0px; overflow: hidden;"
                            />
                            <textarea
                              aria-hidden="true"
                              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
                              readonly=""
                              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
                              tabindex="-1"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-actions-7"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              undo changes
            </span>
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
            style="margin-left: auto;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              cancel
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              apply
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
